edb/schema/inheriting.py

```diff
@@ -514,26 +514,18 @@ def _classbases_from_ast(
 
         return base_refs
 
-    def _apply_field_ast(
+    def get_ast_attr_for_field(
         self,
-        schema: s_schema.Schema,
-        context: sd.CommandContext,
-        node: qlast.DDLOperation,
-        op: sd.AlterObjectProperty,
-    ) -> None:
+        field: str,
+        astnode: Type[qlast.DDLOperation],
+    ) -> Optional[str]:
         if (
-            op.property in {'is_abstract', 'is_final'}
-            and not isinstance(self, sd.DeleteObject)
+            field in {'is_abstract', 'is_final'}
+            and issubclass(astnode, qlast.CreateObject)
```

Comment:
Could some of this kind of logic be made data-driven with field properties?

Reply:
Yes. The plan is to follow this up with some sort of structure that can be specified on the field. Ideally, the info should be bi-directional, i.e. used in `_cmd_tree_from_ast`.  I decided not to do this in this PR, since it's already fairly big and `get_ast_attr_for_field` was already there.

Reply:
:+1: